Tasks/ServiceFabricDeploy/ServiceFabricSDK/Publish-UpgradedServiceFabricApplication.ps1

```diff
@@ -200,9 +203,21 @@ function Publish-UpgradedServiceFabricApplication
 
         $reg = Get-ServiceFabricApplicationType -ApplicationTypeName $names.ApplicationTypeName | Where-Object  { $_.ApplicationTypeVersion -eq $names.ApplicationTypeVersion }
         if ($reg)
-        {
-            Write-Host (Get-VstsLocString -Key SFSDK_UnregisteringExistingAppType -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))
-            $reg | Unregister-ServiceFabricApplicationType -Force
+        {            
+            try {
+                Write-Host (Get-VstsLocString -Key SFSDK_UnregisteringExistingAppType -ArgumentList @($names.ApplicationTypeName, $names.ApplicationTypeVersion))            
+                $reg | Unregister-ServiceFabricApplicationType -Force    
+            }
+            catch [System.Fabric.FabricException] {
+                ## If SkipUpgrade param is set when existing version already deployed, then back out of upgrade
+                if($SkipUpgradeSameTypeAndVersion -And $Error[0].Exception.Message -like '*Application type and version is still in use*') {                    
```

Comment:
True, missed the ErrorCode property before. 

Regarding the enum value it would ideally be ApplicationAlreadyInTargetVersion thrown during upgrade, but the exception occurs prior to the upgrade, instead it is ApplicationTypeInUse thrown when the app type and version is already found and is being unregistered.  

I never truly understood why an app would be unregistered before an upgrade, but thought is was some sort of pre-validation step, which would cause the error to be thrown, hopefully intentionaly. I don't know if the pre-validation step is really needed if the upgrade actally throws ApplicationAlreadyInTargetVersion if it finds the same app type and version as in the upgrade package. Then the catch block could just wrap the upgrade step.